frontend/src/components/pages/RegistrantExperience/PersonalInfo/personalInfoReducer.tsx

```diff
@@ -0,0 +1,166 @@
+        /* eslint-disable-next-line */
+            for (const contact of newCampers[0].contacts){
+          newCampers[newCampers.length - 1].contacts.push(
+            JSON.parse(JSON.stringify(contact)),
+          ); // Deep copy the contact
+        }
+        break;
+      }
+      case PersonalInfoActions.DELETE_CAMPER: {
+        const { camperId } = action as DeleteCamper;
+        newCampers.splice(camperId, 1);
+        break;
+      }
+      case PersonalInfoActions.UPDATE_CAMPER: {
+        const { camperId, field, data } = action as UpdateCamper;
+        if (field === "firstName") newCampers[camperId][field] = data as string;
+        else if (field === "lastName")
+          newCampers[camperId][field] = data as string;
+        else if (field === "age") newCampers[camperId][field] = data as number;
+        else if (field === "allergies")
+          newCampers[camperId][field] = data as string;
+        else if (field === "specialNeeds")
+          newCampers[camperId][field] = data as string;
+        break;
+      }
+      case PersonalInfoActions.UPDATE_CONTACT: {
+        const { contactId, field, data } = action as UpdateContact;
+        /* eslint-disable-next-line */
+        for (const camper of newCampers) {
+          if (field === "firstName")
+            camper.contacts[contactId][field] = data as string;
+          else if (field === "lastName")
+            camper.contacts[contactId][field] = data as string;
+          else if (field === "email")
+            camper.contacts[contactId][field] = data as string;
+          else if (field === "phoneNumber")
+            camper.contacts[contactId][field] = data as string;
+          else if (field === "relationshipToCamper")
+            camper.contacts[contactId][field] = data as string;
+        }
+        break;
+      }
+      default:
+    }
+    return newCampers;
+  });
+};
+
+export const usePersonalInfoHook = (
+  setCampers: React.Dispatch<
+    React.SetStateAction<RegistrantExperienceCamper[]>
+  >,
+): ((action: PersonalInfoReducerDispatch) => void) => {
+  const dispatch = (action: PersonalInfoReducerDispatch) => {
+    CamperReducer(setCampers, action);
+  };
+  return dispatch;
+};
+
+export const checkFirstName = (firstName: string): boolean => {
+  return !!firstName;
+};
+
+export const checkLastName = (lastName: string): boolean => {
+  return !!lastName;
+};
+
+export const checkAge = (age: number): boolean => {
+  return !!age && age >= 7 && age <= 10;
+};
+
+export const checkEmail = (email: string): boolean => {
+  return !!email;
+};
+
+export const checkPhoneNumber = (phoneNumber: string): boolean => {
+  return !!phoneNumber;
+};
+
+export const checkRelationToCamper = (relation: string): boolean => {
+  return !!relation;
+};
+
+export const checkPersonalInfoFilled = (
+  campers: RegistrantExperienceCamper[],
+): boolean => {
+  if (!(campers.length >= 1)) return false;
+  /* eslint-disable-next-line */
+  for (const camper of campers) {
```

Comment:
forEach here?

Reply:
I think you might be able to use the every function too actually

Reply:
Not fully sure if I can use forEach here because I ultimately need to return true or false in the function `checkPersonalInfoFilled`. I think calling `return false` or `return true` in a forEach doesn't actually push true or false to be returned by `checkPersonalInfoFilled`.